extensions/amp-accordion/1.0/base-element.js

```diff
@@ -106,10 +106,12 @@ function getState(element, mu, getExpandStateTrigger) {
     const headerProps = dict({
       'as': headerShim,
       'id': section.firstElementChild.getAttribute('id'),
+      'role': section.firstElementChild.getAttribute('role') || undefined,
     });
     const contentProps = dict({
       'as': contentShim,
       'id': section.lastElementChild.getAttribute('id'),
+      'role': section.lastElementChild.getAttribute('role') || undefined,
```

Comment:
Had to add `|| undefined` because in the case the `role` attribute does not exist, `null` will be passed to `AccordionHeader` and `AccordionContent` preact side components while will be passed into the DOM element resulting in `role='null'`.  `|| undefined` allows for default value in props to be used.

Reply:
This is different from `id` which is has additional logic to handle it and does not rely on prop default values.  If a `null` value is passed into `id` it is treated the same way as `undefined` (or any falsy value).

Reply:
This sounds like a great in-code comment :)

Reply:
Thanks for the comment, added in a comment in the code!